packages/react-conformance/src/utils/getCallbackArguments.test.ts

```diff
@@ -0,0 +1,217 @@
+    //     export interface AccordionProps {
+    //       onToggle: AccordionToggleEventHandler;
+    //     }`,
+    //   });
+    //   const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+    //
+    //   expect(type).toMatchObject({ event: ['React.MouseEvent', 'React.KeyboardEvent'], data: { value: 'String' } });
+    // });
+
+    it('handles generics', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': `import * as React from 'react';
+      
+        export interface AccordionProps { onToggle: (e: React.MouseEvent<HTMLElement>) => void; }`,
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ e: 'React.MouseEvent' });
+    });
+
+    it('handles unions', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': `import * as React from 'react';
+      
+        export interface AccordionProps { onToggle: (e: MouseEvent | React.MouseEvent) => void; }`,
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ e: ['MouseEvent', 'React.MouseEvent'] });
+    });
+
+    it('handles multiple params', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': `import * as React from 'react';
+      
+        export interface AccordionProps { onToggle: (e: null, data: { value: string }) => void; }`,
+      });
+      const type = getCallbackArguments(program, 'Accordion.types.ts', 'AccordionProps', 'onToggle');
+
+      expect(type).toMatchObject({ e: null, data: { value: 'String' } });
+    });
+  });
+
+  describe('errors', () => {
+    it('throws when a file is not found', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': 'export interface AccordionProps { onToggle: (e: null) => void; }',
+      });
+
+      expect(() => getCallbackArguments(program, 'Button.types.ts', 'AccordionProps', 'onToggle')).toThrowError(
+        [
+          'A file (Button.types.ts) was not found in TS program, this looks like an invocation problem,',
+          'check your params',
+        ].join(' '),
+      );
+    });
+
+    it('throws when an interface is not found', async () => {
+      const program = await setupProgram(['Accordion.types.ts'], {
+        './Accordion.types.ts': 'export interface AccordionProps { onToggle: (e: null) => void; }',
+      });
+
+      expect(() =>
+        getCallbackArguments(program, 'Accordion.types.ts', 'ButtonProps', 'onToggle'),
+      ).toThrowErrorMatchingInlineSnapshot(
+        `"A file (Accordion.types.ts) does not contain definition for type \\"ButtonProps\\""`,
+      );
+    });
+
+    it('throws when an interface is not found', async () => {
```

Comment:
This probably should be something like:
```suggestion
    it('throws when a property is not found', async () => {
```